### ISSUE TITLE
museval support

### ISSUE THREAD

**Issue by faroit** [JuanFMontesinos/torch_mir_eval#3]
Hi @JuanFMontesinos. First of all, this is great. I always wanted to do such a port but I never got the time to actually do it. :+1:

Given the wish to remove bsseval from mir_eval (See thread [here](https://github.com/craffel/mir_eval/issues/68#issuecomment-495227366), the idea was to put the separation module into a separate package (basically stripping out the code from museval). I started working on this [here](https://github.com/sigsep/bsseval/) but I didn't really worked on it lately. So here are some questions proposals:

* Do you plan to also add support to v4 (aka. [museval](https://github.com/sigsep/sigsep-mus-eval)?
* Would you be interested in taking over `bsseval`? I would be fine with torch-only actually

Also pinging @mpariente, any ideas?

**Comment by JuanFMontesinos** [comment]
Hi @faroit,

Let me be sincere and talk a little bit about the thread.
I'm actually working on audiovisual BSS.
Discussions about metrics are kind of silent and in petit comité. When people report metrics, you don't really know which functions they used. 

I can say that most of the works I've seen uses `mir_eval` and `bss_eval_sources`. 
In that sense, `museval` is perceived more like a `musdb` toolkit rather than a source separation package (just by reading the description). This is a bit linked to the "findability" of the library which is mentioned in the thread. I think if you aim to create a new library, museval and mir_eval bss modules should be maintained but internally call bss_eval as suggested in the thread. One of the advantages of v3 is it is straightforward to use. In v3 you just need to pass the sources and be happy, in v4 you need to set several parameters.  This may lead to rely on v3 if there is no clear explanation in the docs.

WRT to your questions:
As a community leader I imagine you have a clear idea about are people using and reporting. 
I personally ported (and tested) this  as  a way of pushing the community to port everything else. Soo I wasn't thinking about supporting it unless it is strongly used by the community (but I have no idea about the real state). However, given that v3 version is format is provided too, it can be worth the effort. 
I can take over bsseval. If that's the case it would be nice to have some help.

In short, if the community agrees to merge everything in a single repo and to clarify which metrics to use (through a more detailed explanation in further papers and warnings in mir_eval and museval) it can be a nice idea and ofc I'd be happy to contribute.

**Comment by JuanFMontesinos** [comment]
Hi again @faroit  @mpariente ,
Since Pytorch has released V1.7 which includes a new fft module and full support for complex numbers (they are deprecating the modules used right now). It can be worth to port mus_eval directly instead of mir_eval. 

I will be doing so if you have no plans for developing the bss_eval merge in the short time.

Regards
Juan

**Comment by faroit** [comment]
> In that sense, museval is perceived more like a musdb toolkit rather than a source separation package (just by reading the description). This is a bit linked to the "findability" of the library which is mentioned in the thread. I think if you aim to create a new library, museval and mir_eval bss modules should be maintained but internally call bss_eval as suggested in the thread. One of the advantages of v3 is it is straightforward to use. In v3 you just need to pass the sources and be happy, in v4 you need to set several parameters. This may lead to rely on v3 if there is no clear explanation in the docs.

I agree with your assessment. I think currently the best would be if we would all join forces to create the new `bsseval` package. 

> WRT to your questions:
> As a community leader I imagine you have a clear idea about are people using and reporting.
> I personally ported (and tested) this as a way of pushing the community to port everything else. Soo I wasn't thinking about > supporting it unless it is strongly used by the community (but I have no idea about the real state). However, given that v3 > version is format is provided too, it can be worth the effort.
> I can take over bsseval. If that's the case it would be nice to have some help.

Sure, I maybe we can have a call together with @mpariente and @pseeth ?

> In short, if the community agrees to merge everything in a single repo and to clarify which metrics to use (through a more detailed explanation in further papers and warnings in mir_eval and museval) it can be a nice idea and ofc I'd be happy to contribute.

Yes, I think that would stop the confusion. CPU and GPU implementations for bss_eval v3, v4 and SI-SDR is all we need.

**Comment by mpariente** [comment]
Available for a call !

**Comment by JuanFMontesinos** [comment]
Hi,
I'm available too
